### DISCUSSION TITLE
curtailment_costs not being reported?

### DISCUSSION THREAD

**Discussion by fares2201** [spine-tools/SpineOpt.jl#857] [Support]
Hello,

There seems to be an issue with curtailment_costs not being reported to the total_costs. Ive attached a simple model with a wind unit and electricity node, and its output, in a zip file in this thread. 
[CurtialmentIssue.zip](https://github.com/spine-tools/SpineOpt.jl/files/12359589/CurtialmentIssue.zip)

Looking at total costs, it is simply [the hourly electricity demand (100) * vom_cost of Wind_to_Electricity (2.0) * hourly timesteps for a week (168) = 33,600] negating the curtailment costs I've placed on the wind unit (20). With the wind unit having a capacity of 200, I'd expect curtailment costs to be [(200-100) * 20 * 168 = 336,000] but the objective_renewable_curtailment_cost output yields a value of 0. Am i missing something with my model to make curtailments work or is this a bug? 

FYI, toolbox and SpineOpt are up to the latest versions. 

Thank you,
Fares

**Comment by nnhjy** [comment]
If you set the parameter `online_variable_type` to be `binary` or `integer` for your wind unit, the curtailment cost should be reported as expected.

The problem was that with the default linear variable `units_on`, the model was just solved with the variable being 0.5 to avoid curtailment cost.

**Comment by fares2201** [comment]
Thank you for your fast response. It is indeed fixed now.

**Comment by fares2201** [comment]
Hey, would like to clarify how curtailment works in Spine as Im getting counterintuitive results. Im conducting a model on the GB electricity system with wind included. I attached a screenshot of the wind unit model in this thread. 

The problem is that Im getting certain hourly periods in where the wind unit decides to turn off completely which Im assuming is to avoid curtailment. It turns off completely only during periods when the wind generation is greater than the connection_capacity limit being set. During periods when the wind generation is less than the connection_capacity, it only curtails in very few periods but I assume this is mainly for economical reasons? 

So does SPINE consider that curtailment_costs are avoided when the wind unit shuts is off completely. This goes against my understanding of curtailment and the definition for curtailment_cost in the manual: "...accrues costs whenever unit_flow not operating at its maximum available capacity"

<img width="914" alt="Screen Shot 2023-08-19 at 5 32 22 PM" src="https://github.com/spine-tools/SpineOpt.jl/assets/128616405/6313e37e-b28e-418b-9db2-04a5b4ba581e">

**Comment by clizbe** [comment]
@fares2201 Did you figure this out or do you still need help?
